File: book/스프링부트를 활용한 마이크로서비스 개발/test/src/main/java/com/example/test/controller/MultiplicationResultAttemptController.java
package com.example.test.controller;

import com.example.test.domain.Multiplication;
import com.example.test.domain.MultiplicationResultAttempt;
import com.example.test.service.MultiplicationService;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/results")
final class MultiplicationResultAttemptController {
    private final MultiplicationService multiplicationService;

    @Autowired
    public MultiplicationResultAttemptController(final MultiplicationService multiplicationService) {
        this.multiplicationService = multiplicationService;
    }

    @PostMapping
    ResponseEntity<ResultResponse> postResult(@RequestBody MultiplicationResultAttempt multiplicationResultAttempt){
        return ResponseEntity.ok(
                new ResultResponse(multiplicationService.checkAttempt(multiplicationResultAttempt)));
    }

    @RequiredArgsConstructor
    @NoArgsConstructor(force = true)
    @Getter
    static final class ResultResponse {
        private final boolean correct;
    }
}

File: book/스프링부트를 활용한 마이크로서비스 개발/kotlin/src/main/kotlin/com/example/kotlin/controller/MultiplicationResultAttemptController.kt
package com.example.kotlin.controller

import com.example.kotlin.domain.MultiplicationResultAttempt
import com.example.kotlin.service.MultiplicationService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/results")
class MultiplicationResultAttemptController(
    @Autowired
    val multiplicationService: MultiplicationService
) {
    class ResultResponse(correct: Boolean) {
        var correct: Boolean = true
    }

    @PostMapping
    fun postResult(@RequestBody multiplicationResultAttempt: MultiplicationResultAttempt): ResponseEntity<ResultResponse> {
        return ResponseEntity.ok(ResultResponse(multiplicationService.checkAttempt(multiplicationResultAttempt)))
    }
}